server/game/core/ongoingEffect/OngoingEffect.ts

```diff
@@ -42,54 +64,56 @@ class OngoingEffect {
         this.impl = effectImpl;
         this.ability = properties;
         this.targets = [];
+        this.targetController = properties.targetController || RelativePlayer.Self;
```

Comment:
Wondering if this should be here? `targetController` is referring to the controller of a card which wouldn't be meaningful if the target is a player

Reply:
Oh hmm, let's move it back into just `OngoingCardEffect` and see what happens? Definitely it shouldn't be set this way imo. Just a lot of messiness in the effects in general (as you've seen) so it may have just been a copy-paste error

Reply:
Aaaah ok, so it is used by both, but in the context of OngoingPlayerEffect it means "Target Controller" rather than "Target's Controller". OngoingPlayerEffect can also have it be a literal player object in addition to being a RelativePlayer, whereas OngoingCardEffect can only be a RelativePlayer, at least from what I can infer. I'll move it up and update the types appropriately.

Reply:
Addendum: So it's a little harder to enforce this on the ability creation side, the I added separate interfaces for IPlayerOngoingEffectProps and ICardOngoingEffectProps, but the ability creation process just has it's own set of interfaces it loosely tacks on which include targetController. I've still separated it, but added a enforcement in OngoingCardEffect to ensure it's always a relative player.

Reply:
Ahh okay, finally found an explanation in the comments. The naming for this property is just horrible:
![image](https://github.com/user-attachments/assets/c1bbacb0-dc60-48cc-9f70-fbb7bceba9e3)